File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/StatusBar/Layouts/Docking.vb
Namespace CompanionKit.Controls.StatusBar.Layouts

	Public Class Docking

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        Private Sub Docking_Load(sender As Object, e As EventArgs) Handles Me.Load
            ' Fill up the ComboBox with enable docking styles.
            Dim mobjDefaultDockStatusBar As DockStyle = Me.mobjTestStatusBar.Dock
            Me.mobjDockComboBox.DataSource = [Enum].GetValues(GetType(DockStyle))
            Me.mobjDockComboBox.SelectedItem = mobjDefaultDockStatusBar
        End Sub

        ''' <summary>
        ''' Handles SelectedIndexChanged event of the ComboBox.
        ''' Changes docking style for the StatusBar.
        ''' </summary>
        Private Sub mobjDockComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjDockComboBox.SelectedIndexChanged
            'Change docking style of the StatusBar.
            Me.mobjTestStatusBar.Dock = DirectCast(Me.mobjDockComboBox.SelectedItem, DockStyle)
        End Sub
	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/StatusBar/Layouts/Docking.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.StatusBar.Layouts
{
    public partial class Docking : UserControl
    {
        public Docking()
        {
            InitializeComponent();
        }

        private void Docking_Load(object sender, EventArgs e)
        {
            // Fill up the ComboBox with enable docking styles.
            DockStyle mobjDefaultDockStatusBar = this.mobjTestStatusBar.Dock;
            this.mobjDockComboBox.DataSource = Enum.GetValues(typeof(DockStyle));
            this.mobjDockComboBox.SelectedItem = mobjDefaultDockStatusBar;
        }

        /// <summary>
        /// Handles SelectedIndexChanged event of the ComboBox.
        /// Changes docking style for the StatusBar.
        /// </summary>
        private void mobjDockComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Change docking style of the StatusBar.
            this.mobjTestStatusBar.Dock = (DockStyle)this.mobjDockComboBox.SelectedItem;
        }
    }
}